I fixed the dependency error, but it's now failing with:

```
[INFO] There is 1 error reported by Checkstyle 9.3 with contrib/checkstyle.xml ruleset.
[ERROR] src/main/java/org/apache/accumulo/examples/client/TracingExample.java:[34,1] (imports) ImportControl: Disallowed import - org.apache.accumulo.core.trace.TraceUtil.
```

Is this an issue, or should I fix the checkstyle.xml file?

--- a/README.md
+++ b/README.md
@@ -87,6 +87,7 @@ Each example below highlights a feature of Apache Accumulo.
 | [spark] | Using Accumulo as input and output for Apache Spark jobs |
 | [tabletofile] | Using MapReduce to read a table and write one of its columns to a file in HDFS. |
 | [terasort] | Generating random data and sorting it using Accumulo. |
+| [tracing] | Generating trace data in a client application and Accumulo. |
 | [uniquecols] | Use MapReduce to count unique columns in Accumulo |
 | [visibility] | Using visibilities (or combinations of authorizations). Also shows user permissions. |
 | [wordcount] | Use MapReduce and Accumulo to do a word count on text files |
@@ -125,6 +126,7 @@ This repository can be used to test Accumulo release candidates.  See
 [spark]: spark/README.md
 [tabletofile]: docs/tabletofile.md
 [terasort]: docs/terasort.md
+[tracing]: docs/tracing.md
 [uniquecols]: docs/uniquecols.md
 [visibility]: docs/visibility.md
 [wordcount]: docs/wordcount.md

--- a/bin/runex
+++ b/bin/runex
@@ -41,5 +41,5 @@ fi
 "$ex_home"/bin/build
 
 export CLASSPATH="$EXAMPLES_JAR_PATH:$ex_home/conf:$ACCUMULO_HOME/lib/*:$CLASSPATH"
-java "org.apache.accumulo.examples.$1" ${*:2}
+java ${ACCUMULO_JAVA_OPTS} -Dlog4j.configurationFile="$ex_home"/conf/log4j2.properties "org.apache.accumulo.examples.$1" ${*:2}
 

--- a/bin/runex
+++ b/bin/runex
@@ -41,5 +41,5 @@ fi
 "$ex_home"/bin/build
 
 export CLASSPATH="$EXAMPLES_JAR_PATH:$ex_home/conf:$ACCUMULO_HOME/lib/*:$CLASSPATH"
-java ${ACCUMULO_JAVA_OPTS} -Dlog4j.configurationFile="$ex_home"/conf/log4j2.properties "org.apache.accumulo.examples.$1" ${*:2}
+java ${ACCUMULO_JAVA_OPTS} "org.apache.accumulo.examples.$1" ${*:2}
 

--- a/docs/tracing.md
+++ b/docs/tracing.md
@@ -1,0 +1,41 @@
+<!--
+Licensed to the Apache Software Foundation (ASF) under one or more
+contributor license agreements.  See the NOTICE file distributed with
+this work for additional information regarding copyright ownership.
+The ASF licenses this file to You under the Apache License, Version 2.0
+(the "License"); you may not use this file except in compliance with
+the License.  You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+-->
+# Apache Accumulo Tracing Example
+
+This tutorial uses the [TracingExample.java] Java class to create an OpenTelemetry
+span in the TracingExample application and then create and read entries from Accumulo.
+Tracing output should show up in logs for the TracingExample class and the Accumulo client,
+and logs for the Accumulo server processes.
+
+## Setup
+
+  1. Download the OpenTelemetry [JavaAgent] jar file and place it into the `/path/to/accumulo/lib/` directory.
+  2. Add the property `general.opentelemetry.enabled=true` to accumulo.properties
+  3. Set the following environment variables in your environment:
+  
+    ACCUMULO_JAVA_OPTS="-javaagent:/path/to/accumulo/lib/opentelemetry-javaagent-1.12.1.jar"
+    OTEL_TRACES_EXPORTER="logging"
+
+## Run the Example
+
+  1. Start Accumulo
+  2. Run the Tracing Example:
+  
+    $ ./bin/runex client.TracingExample --createtable --deletetable --create --read --table traceTest
+
+[JavaAgent]: https://search.maven.org/remotecontent?filepath=io/opentelemetry/javaagent/opentelemetry-javaagent/1.12.1/opentelemetry-javaagent-1.12.1.jar
+[TracingExample.java]: ../src/main/java/org/apache/accumulo/examples/client/TracingExample.java

--- a/src/main/java/org/apache/accumulo/examples/client/TracingExample.java
+++ b/src/main/java/org/apache/accumulo/examples/client/TracingExample.java
@@ -31,8 +31,6 @@ import org.apache.accumulo.core.data.Key;
 import org.apache.accumulo.core.data.Mutation;
 import org.apache.accumulo.core.data.Value;
 import org.apache.accumulo.core.security.Authorizations;
-import org.apache.accumulo.core.trace.TraceUtil;
-// import org.apache.accumulo.core.trace.DistributedTrace;
 import org.apache.accumulo.examples.Common;
 import org.apache.accumulo.examples.cli.ClientOnDefaultTable;
 import org.apache.accumulo.examples.cli.ScannerOpts;
@@ -41,7 +39,9 @@ import org.slf4j.LoggerFactory;
 
 import com.beust.jcommander.Parameter;
 
+import io.opentelemetry.api.GlobalOpenTelemetry;
 import io.opentelemetry.api.trace.Span;
+import io.opentelemetry.api.trace.Tracer;
 import io.opentelemetry.context.Scope;
 
 /**
@@ -53,6 +53,7 @@ public class TracingExample {
   private static final String DEFAULT_TABLE_NAME = "test";
 
   private final AccumuloClient client;
+  private final Tracer tracer;
 
   static class Opts extends ClientOnDefaultTable {
     @Parameter(names = {"--createtable"}, description = "create table before doing anything")
@@ -72,30 +73,33 @@ public class TracingExample {
 
   private TracingExample(AccumuloClient client) {
     this.client = client;
-  }
-
-  private void enableTracing() {
-    TraceUtil.enable();
+    this.tracer = GlobalOpenTelemetry.get().getTracer(TracingExample.class.getSimpleName());
   }
 
   private void execute(Opts opts) throws TableNotFoundException, AccumuloException,
       AccumuloSecurityException, TableExistsException {
 
-    if (opts.createtable) {
-      Common.createTableWithNamespace(client, opts.getTableName());
+    Span span = tracer.spanBuilder("trace example").startSpan();
+    try (Scope scope = span.makeCurrent()) {
+      if (opts.createtable) {
+        Common.createTableWithNamespace(client, opts.getTableName());
+      }
+
+      if (opts.createEntries) {
+        createEntries(opts);
+      }
+
+      if (opts.readEntries) {
+        readEntries(opts);
+      }
+
+      if (opts.deletetable) {
+        client.tableOperations().delete(opts.getTableName());
+      }
+    } finally {
+      span.end();
     }
 
-    if (opts.createEntries) {
-      createEntries(opts);
-    }
-
-    if (opts.readEntries) {
-      readEntries(opts);
-    }
-
-    if (opts.deletetable) {
-      client.tableOperations().delete(opts.getTableName());
-    }
   }
 
   private void createEntries(Opts opts) throws TableNotFoundException, AccumuloException {
@@ -104,7 +108,7 @@ public class TracingExample {
     // the write operation as it is occurs asynchronously. You can optionally create additional
     // Spans
     // within a given Trace as seen below around the flush
-    Span span = TraceUtil.startSpan(TracingExample.class, "createEntries");
+    Span span = tracer.spanBuilder("createEntries").startSpan();
     try (Scope scope = span.makeCurrent()) {
       try (BatchWriter batchWriter = client.createBatchWriter(opts.getTableName())) {
         Mutation m = new Mutation("row");
@@ -125,7 +129,7 @@ public class TracingExample {
 
     try (Scanner scanner = client.createScanner(opts.getTableName(), opts.auths)) {
       // Trace the read operation.
-      Span span = TraceUtil.startSpan(TracingExample.class, "createEntries");
+      Span span = tracer.spanBuilder("readEntries").startSpan();
       try (Scope scope = span.makeCurrent()) {
         int numberOfEntriesRead = 0;
         for (Entry<Key,Value> entry : scanner) {
@@ -147,7 +151,6 @@ public class TracingExample {
 
     try (AccumuloClient client = opts.createAccumuloClient()) {
       TracingExample tracingExample = new TracingExample(client);
-      tracingExample.enableTracing();
       tracingExample.execute(opts);
     } catch (Exception e) {
       log.error("Caught exception running TraceExample", e);

